include/lsst/afw/image/MaskedImage.h

```diff
@@ -79,20 +79,17 @@ class MaskedImage : public lsst::daf::base::Persistable,
                     public lsst::daf::base::Citizen {
 public:
     /// shared pointer to the Image
-    typedef typename Image<ImagePixelT>::Ptr ImagePtr;
+    typedef std::shared_ptr<image::Image<ImagePixelT>> ImagePtr;
     /// shared pointer to the Mask
-    typedef typename Mask<MaskPixelT>::Ptr MaskPtr;
+    typedef std::shared_ptr<image::Mask<MaskPixelT>> MaskPtr;
     /// shared pointer to the variance Image
-    typedef typename Image<VariancePixelT>::Ptr VariancePtr;
-    /// shared pointer to a MaskedImage
-    typedef std::shared_ptr<MaskedImage> Ptr;
-    typedef std::shared_ptr<const MaskedImage> ConstPtr;
+    typedef std::shared_ptr<image::Image<VariancePixelT>> VariancePtr;
```

Comment:
That seems reasonable to me. We aren't using them widely, but are making some use of them, and it's simpler than having to use a pixel type to construct a pointer type.